core/src/test/java/fr/sncf/osrd/api/pathfinding/PathfindingElectrificationTest.java

```diff
@@ -0,0 +1,187 @@
+package fr.sncf.osrd.api.pathfinding;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import com.google.common.collect.Range;
+import fr.sncf.osrd.Helpers;
+import fr.sncf.osrd.api.ApiTest;
+import fr.sncf.osrd.api.pathfinding.PathfindingRoutesEndpoint;
+import fr.sncf.osrd.api.pathfinding.request.PathfindingWaypoint;
+import fr.sncf.osrd.api.pathfinding.response.NoPathFoundError;
+import fr.sncf.osrd.infra.api.Direction;
+import fr.sncf.osrd.infra.api.signaling.SignalingInfra;
+import fr.sncf.osrd.infra.api.signaling.SignalingRoute;
+import fr.sncf.osrd.infra.api.tracks.undirected.TrackSection;
+import fr.sncf.osrd.railjson.schema.common.graph.EdgeDirection;
+import fr.sncf.osrd.train.TestTrains;
+import fr.sncf.osrd.utils.graph.Pathfinding;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.stream.Stream;
+
+
+public class PathfindingElectrificationTest extends ApiTest {
+    private static void fillInfraWithCatenaries(SignalingInfra infra, String voltage) {
+        for (var track : infra.getTrackGraph().edges()) {
+            if (track instanceof TrackSection)
+                track.getVoltages().put(Range.closed(0., track.getLength()), voltage);
+        }
+    }
+
+    private static void removeMiddleCatenary(Pathfinding.Result<SignalingRoute> normalPath) {
+        var middleRoute = normalPath.ranges().get(normalPath.ranges().size() / 2);
+        var tracks = middleRoute.edge().getInfraRoute().getTrackRanges();
+        var middleTrack = tracks.get(tracks.size() / 2).track.getEdge();
+        middleTrack.getVoltages().put(Range.closed(middleTrack.getLength() * 0.1, middleTrack.getLength() * 0.9), "");
+    }
+
+    private static void addMiddleDeadSection(Pathfinding.Result<SignalingRoute> normalPath, Direction direction) {
+        var middleRoute = normalPath.ranges().get(normalPath.ranges().size() / 2);
+        var tracks = middleRoute.edge().getInfraRoute().getTrackRanges();
+        var middleTrack = tracks.get(tracks.size() / 2).track.getEdge();
+        middleTrack.getDeadSections(direction).add(Range.closed(0., middleTrack.getLength() * 0.9));
+    }
```

Comment:
There's some duplicated code here, and it seems to be important that it stays identical. It could be moved to a small function